perfkitbenchmarker/analyze.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Functions for analyzing output from benchmarks.
+
+In general, this module works on data in the form of Pandas data
+frames, because that seems to be the most standard Python statistical
+data format.
+"""
+
+import pandas as pd
+
+from perfkitbenchmarker import sample
+
+
+def AllStreamsInterval(start_times, durations, stream_ids):
+  """Compute when all streams were active from multistream records.
+
+  Args:
+    start_times: a pd.Series of POSIX timestamps, as floats.
+    durations: a pd.Series of durations, as floats measured in seconds.
+    stream_ids: a pd.Series of any type.
+
+  All three arguments must have matching indices. Each (start_time,
```

Comment:
add some asserts at the beginning of this function to check this. 

(if this condition is not met, wrong results would be produced, I'd rather see our program crash/assert than reporting wrong results)